### PR TITLE
Added NVML bindings.

### LINKED ISSUE #387: Feature request: Add the ability to receive data from gpu sensors
I propose to add the ability to add methods that return the current parameters of the accelerator:
* temperature
* fan speed
* cpu and memory frequencies

### PR DESCRIPTION
**Pull request by MoFtZ**
Fixes #387.

Added NVML bindings for:
- Device Queries
- System Queries
- Unit Queries

Also added `NvmlDevice` helper class to make it easier to interact with common Device Queries. Note that not all Device Query methods are directly mapped in `NvmlDevice`. To access these other methods, use the `API` property of `NvmlDevice`, or directly create an `NvmlAPI` instance. This `NvmlAPI` instance is also required when calling System and Unit Queries.
```CSharp
// NvmlDevice wrapper
var device = NvmlDevice.CreateFromAccelerator(accelerator);
var temp = device.GetGpuTemperature();

// Calling NvmlAPI directly
device.API.DeviceGetTemperature(
        device.DeviceHandle,
        NvmlTemperatureSensors.NVML_TEMPERATURE_GPU,
        out temp));

// Create separate NvmlAPI instance
var nvml = NvmlAPI.Create(NvmlAPIVersion.V6);
nvml.UnitGetCount(out uint count);
```

### CODE REVIEW

File: Src/ILGPU.Algorithms/Runtime/Cuda/API/NvmlAPI.xml [hunk 1]
```diff
@@ -0,0 +1,609 @@
+﻿<?xml version="1.0" encoding="utf-8" ?>
+<ImportLibrary Namespace="ILGPU.Runtime.Cuda.API"
+               ClassName="NvmlAPI"
+               VersionType="NvmlAPIVersion"
+               DefaultMethodAccessModifier="public"
+               DefaultReturnType="NvmlReturn">
+    <LibraryVersions>
+        <LibraryVersion Name="V6" Platform="Windows" LibName="nvml" />
```
Comment (MoFtZ) [inline Src/ILGPU.Algorithms/Runtime/Cuda/API/NvmlAPI.xml:8]: QUESTION: Is V6 the correct version number?
This [link](https://developer.nvidia.com/nvidia-management-library-nvml) only lists "Latest Production" and "Released with CUDA 5.5".

File: Src/ILGPU.Algorithms/Runtime/Cuda/API/NvmlAPI.xml [hunk 2]
```diff
@@ -0,0 +1,609 @@
...
+        <LibraryVersion Name="V6" Platform="Linux" LibName="nvml.so" />
+        <LibraryVersion Name="V6" Platform="OSX" LibName="nvml.dylib" />
```
Comment (MoFtZ) [inline Src/ILGPU.Algorithms/Runtime/Cuda/API/NvmlAPI.xml:10]: QUESTION: Is `nvml.so` and `nvml.dylib` the correct file names for Linux and OSX?

File: Src/ILGPU.Algorithms/Runtime/Cuda/API/NvmlAPI.xml (hunk 1 above)
Comment (m4rs-mt) [reply Src/ILGPU.Algorithms/Runtime/Cuda/API/NvmlAPI.xml:8]: Seems to work without any issues on an NVIDIA DGX machine.

File: Src/ILGPU.Algorithms/Runtime/Cuda/API/NvmlAPI.xml (hunk 2 above)
Comment (m4rs-mt) [reply Src/ILGPU.Algorithms/Runtime/Cuda/API/NvmlAPI.xml:10]: The library bindings need to be adjusted:

- Windows: fine 👍 
- Linux: `nvidia-ml.so`
- OSX: `nvidia-ml.dylib`

**Review by m4rs-mt — CHANGES_REQUESTED** [review]
The library bindings for Linux and OSX must be adapted.

File: Src/ILGPU.Algorithms/Runtime/Cuda/API/NvmlAPI.xml (hunk 2 above)
Comment (MoFtZ) [reply Src/ILGPU.Algorithms/Runtime/Cuda/API/NvmlAPI.xml]: Thanks - bindings adjusted.

**Review by m4rs-mt — APPROVED** [review]

**Pull request merged**